src/action.ts

```diff
@@ -78,6 +78,46 @@ export const runAction = async (space): Promise<void> => {
     },
   };
 
+  if (CREATE_CDA_TOKEN) {
+    const branchName = branchNames.headRef;
+    const keyName = `ephemeral-token-${branchName}`;
+
+    const spaceKeys = await space.getApiKeys();
+
+    let exists = false;
+
+    for(let i=0; i<spaceKeys.items.length; i++){
+      if (spaceKeys.items[i].name == keyName) {
+        exists = true;
+        break;
+      }
+    }
+
+    if (exists) {
+      Logger.log(`CDA token ${keyName} is already created`);
+    } else {
+      Logger.log(`Creating new CDA token "${keyName}"for ephemeral environment "${environmentId}"...`);
+
+      // just create api key with name
+      // environment is always being added to all space keys in the next step
+      space.createApiKey({
+        name: keyName,
+        environments: [newEnv],
+      }).then(key => {
+        core.setOutput(
+          "cda_token",
+          key.accessToken
+        );
+        Logger.success("CDA token has been created");
+      }).catch(err => {
+        Logger.warn("unable to create ephemeral token");
+        Logger.verbose(err)
+      })
```

Comment:
The modern JS for the async/await here is something like

```suggestion
      try {
        const key = await space.createApiKey({
          name: keyName,
          environments: [newEnv],
        })
      
        core.setOutput(
          "cda_token",
          key.accessToken
        );
        Logger.success("CDA token has been created");
      } catch(err) {
        Logger.warn("unable to create ephemeral token");
        Logger.verbose(err)
      }
```